src/libcephsqlite.cc

```diff
@@ -807,8 +819,15 @@ LIBCEPHSQLITE_API int sqlite3_cephsqlite_init(sqlite3* db, char** err, const sql
     vfs->xAccess = Access;
     vfs->xFullPathname = FullPathname;
     vfs->xCurrentTimeInt64 = CurrentTime;
-    appd->cct = nullptr;
-    sqlite3_vfs_register(vfs, 0);
+    if (int rc = sqlite3_vfs_register(vfs, 0); rc) {
+      delete appd;
+      free(vfs);
+      return rc;
+    }
+  }
+
+  if (int rc = std::atexit(cephsqlite_atexit); rc) {
```

Comment:
probably a better way to clean up app data is to hook a function to `xDlClose` in `sqlite3_cephsqlite_init()`?

Reply:
sqlite does not do any cleanup on error, it just calls `exit(3)`.

Reply:
`xDlClose` would be called in the UNIX VFS, in any case, as that's the VFS which loaded `libcephsqlite`.